core/src/Cuda/Kokkos_Cuda_Parallel_Team.hpp

```diff
@@ -872,9 +873,9 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
 #endif
                                  !std::is_same<ReducerType, InvalidType>::value;
     if (!is_empty_range || need_device_set) {
-      const int block_count =
-          UseShflReduction ? std::min(m_league_size, size_type(1024 * 32))
-                           : std::min(int(m_league_size), m_team_size);
+      const int block_count = std::max(
+          1u, UseShflReduction ? std::min(m_league_size, size_type(1024 * 32))
+                               : std::min(int(m_league_size), m_team_size));
```

Comment:
Yuk
I had to stare at it for a minute to see that the ternary uses the common type between `int` and `unsigned int` which is indeed unsigned...

Reply:
I just did what the compiler wanted me to do. :slightly_smiling_face: